### ISSUE TITLE
no-callback-in-async-function rule?

### ISSUE THREAD

**Issue by xjamundx** [xjamundx/eslint-plugin-promise#155]
Putting callback code inside of async functions like this can lead to a lot of trouble:

```js
async function bigProbs() {
    doSomething(function(err) {
        if (err) throw err; // won't be caught by the parent
        // ...
    })
}
```

**Comment by Lonniebiz** [comment]
Would something like this address the issue:
async function bigProbs() {
```
	try
	{
		doSomething(function(err) {
			if (err) throw err; // won't be caught by the parent
			// ...
		})
	}
	catch(e)
	{
		throw(e);
	}
}
If so, can you not warn when the try catch is throwing to the parent?
```

**Comment by brettz9** [comment]
> Would something like this address the issue:
> 
> ```js
> async function bigProbs() {
> 	try
> 	{
> 		doSomething(function(err) {
> 			if (err) throw err; // won't be caught by the parent
> 			// ...
> 		})
> 	}
> 	catch(e)
> 	{
> 		throw(e);
> 	}
> }
> ```
> 
> If so, can you not warn when the try catch is throwing to the parent?

No, your example is equivalent.

I think the intent is to avoid confusion by preventing what could admittedly even be synchronous callbacks (whose errors could actually work here). But it avoids confusion for cases where it really is asynchronous and then the error won't be caught.
